File: java/bypstest-api/src/byps/test/api/srvr/ClientIF.java
package byps.test.api.srvr;

import java.io.InputStream;
import java.util.List;

import byps.BRemote;
import byps.RemoteException;

/**
 * 
 * @BClientRemote
 *
 */
public interface ClientIF extends BRemote {
	
	public int incrementInt(int a) throws RemoteException;

	public List<InputStream> getStreams(int ctrl) throws RemoteException;
	
	public void putStreams(List<InputStream> strm, int ctrl) throws RemoteException;
	
	public ChatStructure sendChat(ChatStructure cs) throws RemoteException;
}

File: csharp/bypstest-ser/src-ser/byps/test/api/srvr/ClientIF.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using byps;

namespace byps.test.api.srvr
{
	
	// checkpoint byps.gen.cs.GenRemoteClass:93
	public interface ClientIF : BRemote {
	
		int IncrementInt(int a);
		void IncrementInt(int a, BAsyncResult<int> asyncResult) ;
		Task<int> IncrementIntAsync(int a);
		
		IList<System.IO.Stream> GetStreams(int ctrl);
		void GetStreams(int ctrl, BAsyncResult<IList<System.IO.Stream>> asyncResult) ;
		Task<IList<System.IO.Stream>> GetStreamsAsync(int ctrl);
		
		void PutStreams(IList<System.IO.Stream> strm, int ctrl);
		void PutStreams(IList<System.IO.Stream> strm, int ctrl, BAsyncResult<Object> asyncResult) ;
		Task PutStreamsAsync(IList<System.IO.Stream> strm, int ctrl);
		
		ChatStructure SendChat(ChatStructure cs);
		void SendChat(ChatStructure cs, BAsyncResult<ChatStructure> asyncResult) ;
		Task<ChatStructure> SendChatAsync(ChatStructure cs);
		
		
	}
}  // end namespace
